examples/metrics/exemplar/src/main/java/io/helidon/examples/metrics/exemplar/GreetService.java

```diff
@@ -154,11 +145,11 @@ private void updateGreetingHandler(ServerRequest request,
     }
 
     private void timeGet(ServerRequest request, ServerResponse response) {
-        timerForGets.time((Runnable) response::next);
+        timerForGets.record((Runnable) response::next);
```

Comment:
`record(Callable)` used to be called `recordCallable` to avoid problems like this. (I had originally followed the naming that Micrometer uses on their `Timer` class). Then in this https://github.com/helidon-io/helidon/pull/7261#discussion_r1279388991 review comment you asked to change it to `record` and I agreed. But with that change `record(Supplier<T>)` and `record(Callable<T>)` have the same erasure.

Which way do you want to go? Change the method name back to `recordCallable` or require casting to disambiguate?

Reply:
actually the problem is not between `Runnable` and `Callable`, but `Callable` and `Supplier`. `response::next` returns a value.
So casting it to `Runnable` is not the best option here.
So the options I see:
- only keep `record(Callable)` and remove `record(Supplier)` (as callable allows for checked exceptions, on the other hand you throw a checked exception from the method as well, which is not according to our guidelines...)
- only keep `record(Supplier)` to allow the same features, but force the user to resolve checked exceptions on their own

I would prefer not to have signatures that conflict where we expect method references to be used.

Reply:
The `Exception` comes from invoking the Micrometer delegate timer's method.

What about keeping `record(Callable)`, catching and wrapping the `Exception` in a `RuntimeException`, and removing `record(Supplier)`?